controllers/ingressClassCreator_test.go

```diff
@@ -12,19 +15,39 @@ import (
 )
 
 func TestNewIngressClassCreator(t *testing.T) {
+	namespace := "test"
 	clientMock := testclient.NewClientBuilder().WithScheme(getScheme()).Build()
-	creator := NewIngressClassCreator(clientMock, "my-ingress-class")
+	creator := NewIngressClassCreator(clientMock, "my-ingress-class", namespace, mocks.NewEventRecorder(t))
 
 	require.NotNil(t, creator)
 }
 
 func TestIngressClassCreator_CreateIngressClass(t *testing.T) {
+	namespace := "test"
+	t.Run("failed to get deployment", func(t *testing.T) {
+		t.Parallel()
+
+		// given
+		clientMock := testclient.NewClientBuilder().WithScheme(getScheme()).WithObjects().Build()
+		creator := NewIngressClassCreator(clientMock, "my-ingress-class", namespace, nil)
+
+		// when
+		err := creator.CreateIngressClass(context.Background())
+
+		// then
+		require.Error(t, err)
+		require.ErrorContains(t, err, "create ingress class: failed to get deployment [k8s-service-discovery-controller-manager]")
```

Comment:
`ErrorContains` also includes and `Error` check. 

```suggestion
		require.ErrorContains(t, err, "create ingress class: failed to get deployment [k8s-service-discovery-controller-manager]")
```